app/Models/Beatmapset.php

```diff
@@ -700,6 +702,30 @@ public function love(User $user)
         ];
     }
 
+    public function rank()
+    {
+        if (!$this->isQualified()) {
+            return false;
+        }
+
+        DB::transaction(function () {
+            $this->events()->create(['type' => BeatmapsetEvent::RANK]);
+
+            $this->setApproved('ranked', null);
```

Comment:
should also delete scores (maybe user ratings too? not sure what currently happens to those on rank)

Reply:
already handled in setApproved method.